### PR TITLE
docs: create a page to show all props used in current code

### PR DESCRIPTION
**Pull request by andreapiras**
## 🖼 Context

We need a list of all props to understand how we can normalize naming across component's API

## 🚀 Changes

 
- Create a hidden page /devs/PropsAPI to list all props used.
- Cleaned up logic inside for props table generator

## ✅ Checklist

- [ ] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/docs/components/Layout.js [hunk 1]
```diff
@@ -30,15 +30,17 @@ const Layout = ({ children, router: { pathname } }) => {
...
-          <li key={section.title}>
-            <Link href={getFirstPageInSection(section).path}>
-              <a className={`Nav__Link ${rootDir === section.dir ? 'active' : ''}`}>
-                {section.title}
-              </a>
-            </Link>
-          </li>
-        ))}
+        {sections
+          .filter((sec) => !sec.hidden)
```
Comment (andreapiras) [inline packages/docs/components/Layout.js:34]: add link to sidebar exclude the hidden ones

File: packages/docs/utils/generatePropsTable/GeneratePropsTable.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+import propsData from '../../temp-tabular-props';
 
 const GeneratePropsTable = ({ componentName }) => {
-  if (!propsData || !propsData[componentName]) {
-    return null;
-  }
+  let data = propsData;
 
-  const schema = propsData[componentName];
+  if (componentName !== '*') {
```
Comment (helendurranttw) [inline packages/docs/utils/generatePropsTable/GeneratePropsTable.js:10]: Might be useful to explain this

File: packages/docs/utils/generatePropsTable/GeneratePropsTable.js [hunk 3]
```diff
@@ -18,25 +18,33 @@ const GeneratePropsTable = ({ componentName }) => {
...
         <tbody>
-          {Object.keys(schema.props).map((propName) => {
-            const prop = schema.props[propName];
-            const { type, allowedValues, required, defaultValue } = parsePropsForTable(prop);
-
-            return (
-              <tr key={propName}>
+          {getPropsList(data).map(
+            ({
+              propName,
```
Comment (helendurranttw) [inline packages/docs/utils/generatePropsTable/GeneratePropsTable.js:32]: Could we alphabetise pleeeeeease 🙃

File: packages/docs/utils/generatePropsTable/GeneratePropsTable.js [hunk 4]
```diff
@@ -18,25 +18,33 @@ const GeneratePropsTable = ({ componentName }) => {
...
+              type,
+              allowedValues,
+              required,
+              defaultValue,
+              description,
+              displayName,
+            }) => (
+              <tr key={`${propName}-${Math.random()}`}>
```
Comment (helendurranttw) [inline packages/docs/utils/generatePropsTable/GeneratePropsTable.js:40]: Just a little bit dodgy. Could keep a count instead?

File: packages/docs/utils/generatePropsTable/GeneratePropsTable.js [hunk 5]
```diff
@@ -46,15 +54,19 @@ const GeneratePropsTable = ({ componentName }) => {
...
+            ),
+          )}
         </tbody>
       </table>
     </div>
   );
 };
 
+const getPropsList = (data) =>
+  data ? Object.values(data).map((prop) => parsePropsForTable(prop)) : [];
```
Comment (helendurranttw) [inline packages/docs/utils/generatePropsTable/GeneratePropsTable.js:68]: You can simplify to `.map(parsePropsForTable)`

File: packages/docs/utils/parsePropsForTable/ParsePropsForTable.js [hunk 6]
```diff
@@ -9,7 +9,14 @@ const cleanValue = (value) => {
   return newValue && typeof newValue === 'string' ? newValue.replace(/"|'/gi, '') : newValue;
 };
 
-const parsePropsForTable = ({ type, required, defaultValue }) => {
+const parsePropsForTable = ({
+  propName,
```
Comment (helendurranttw) [inline packages/docs/utils/parsePropsForTable/ParsePropsForTable.js:13]: A - Z 🙏🙃

**Review by dozortsev — APPROVED** [review]

**Pull request merged**